main.py

```diff
@@ -3,46 +3,22 @@
 import os.path
 import yfinance as yf
 import pandas as pd
+pd.options.mode.chained_assignment = None  # default='warn'
 import numpy as np
 import scipy as cp
 
+import data_fetcher
+
 # %%
-def get_data():
-    # Create an empty list to store the data for each ticker
+def get_data(metrics: list):
+    fetcher = data_fetcher.DataFetcher("downloaded_data")
+    fetcher.init_downloads_directory()
+    for ticker in tickers:
+        fetcher.download_ticker_data(ticker)
+
     ticker_data = []
-    cnt = 0
-    # Loop through each ticker and retrieve the data for the specified metrics
     for ticker in tickers:
-        cnt += 1
-        # Download the ticker data using yfinance
-        print("downloading ticker data for '" + ticker + "' - " + str(cnt) + "/" + str(len(tickers)))
-        ticker_info = yf.Ticker(ticker)
-
-        # Get the historical stock prices for the start and end dates
-        stock_data = yf.Ticker(ticker).history(start=start_date, end=end_date)
-        # Calculate the stock growth in price
-        start_price = stock_data['Close'][0]
-        end_price = stock_data['Close'][-1]
-        # growth = (end_price - start_price) / start_price * 100
-        # print(f"The stock price of {ticker} has grown by {growth:.2f}% over the past 10 years. start_price={start_price:.2f}, end_price={end_price:.2f}")
-
-        # Extract the data for the specified metrics and store it in a dictionary
-        data = {}
-        for metric in metrics:
-            try:
-                value = ticker_info.info[metric]
-            except:
-                value = 'N/A'
-            data[metric] = value
-        # with open("data.json", 'w', encoding='utf-8') as f:
-        #     json.dump(ticker_info.info, f, ensure_ascii=False, indent=2)
-
-        data['price_today'] = end_price
-        data['price_10_years_ago'] = start_price
-        data['growth'] = (end_price - start_price) / start_price * 100
-
-        # Add the ticker's data to the list of data for all tickers
-        ticker_data.append(data)
+        ticker_data.append(fetcher.get_ticker_info(ticker, metrics))
 
     # Convert the list of dictionaries to a Pandas DataFrame
     df = pd.DataFrame(ticker_data, index=tickers)
```

Comment:
should replace all df with something more informative

Reply:
I would usually agree on something like this but df is common for pandas dataframe
anyway, I think we would refactor all of this analyzer section soon